File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-eq-shared-5-4/ReturnLegValuation.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class ReturnLegValuation : ISerialized
    {
        public ReturnLegValuation(XmlNode xmlNode)
        {
            XmlNode initialPriceNode = xmlNode.SelectSingleNode("initialPrice");
            
            if (initialPriceNode != null)
            {
                if (initialPriceNode.Attributes["href"] != null || initialPriceNode.Attributes["id"] != null) 
                {
                    if (initialPriceNode.Attributes["id"] != null) 
                    {
                        initialPriceIDRef_ = initialPriceNode.Attributes["id"].Value;
                        ReturnLegValuationPrice ob = new ReturnLegValuationPrice(initialPriceNode);
                        IDManager.SetID(initialPriceIDRef_, ob);
                    }
                    else if (initialPriceNode.Attributes["href"] != null)
                    {
                        initialPriceIDRef_ = initialPriceNode.Attributes["href"].Value;
                    }
                    else
                    {
                        initialPrice_ = new ReturnLegValuationPrice(initialPriceNode);
                    }
                }
                else
                {
                    initialPrice_ = new ReturnLegValuationPrice(initialPriceNode);
                }
            }
            
        
            XmlNode notionalResetNode = xmlNode.SelectSingleNode("notionalReset");
            
            if (notionalResetNode != null)
            {
                if (notionalResetNode.Attributes["href"] != null || notionalResetNode.Attributes["id"] != null) 
                {
                    if (notionalResetNode.Attributes["id"] != null) 
                    {
                        notionalResetIDRef_ = notionalResetNode.Attributes["id"].Value;
                        XsdTypeBoolean ob = new XsdTypeBoolean(notionalResetNode);
                        IDManager.SetID(notionalResetIDRef_, ob);
                    }
                    else if (notionalResetNode.Attributes["href"] != null)
                    {
                        notionalResetIDRef_ = notionalResetNode.Attributes["href"].Value;
                    }
                    else
                    {
                        notionalReset_ = new XsdTypeBoolean(notionalResetNode);
                    }
                }
                else
                {
                    notionalReset_ = new XsdTypeBoolean(notionalResetNode);
                }
            }
            
        
            XmlNode valuationPriceInterimNode = xmlNode.SelectSingleNode("valuationPriceInterim");
            
            if (valuationPriceInterimNode != null)
            {
                if (valuationPriceInterimNode.Attributes["href"] != null || valuationPriceInterimNode.Attributes["id"] != null) 
                {
                    if (valuationPriceInterimNode.Attributes["id"] != null) 
                    {
                        valuationPriceInterimIDRef_ = valuationPriceInterimNode.Attributes["id"].Value;
                        ReturnLegValuationPrice ob = new ReturnLegValuationPrice(valuationPriceInterimNode);
                        IDManager.SetID(valuationPriceInterimIDRef_, ob);
                    }
                    else if (valuationPriceInterimNode.Attributes["href"] != null)
                    {
                        valuationPriceInterimIDRef_ = valuationPriceInterimNode.Attributes["href"].Value;
                    }
                    else
                    {
                        valuationPriceInterim_ = new ReturnLegValuationPrice(valuationPriceInterimNode);
                    }
                }
                else
                {
                    valuationPriceInterim_ = new ReturnLegValuationPrice(valuationPriceInterimNode);
                }
            }
            
        
            XmlNode valuationPriceFinalNode = xmlNode.SelectSingleNode("valuationPriceFinal");
            
            if (valuationPriceFinalNode != null)
            {
                if (valuationPriceFinalNode.Attributes["href"] != null || valuationPriceFinalNode.Attributes["id"] != null) 
                {
                    if (valuationPriceFinalNode.Attributes["id"] != null) 
                    {
                        valuationPriceFinalIDRef_ = valuationPriceFinalNode.Attributes["id"].Value;
                        ReturnLegValuationPrice ob = new ReturnLegValuationPrice(valuationPriceFinalNode);
                        IDManager.SetID(valuationPriceFinalIDRef_, ob);
                    }
                    else if (valuationPriceFinalNode.Attributes["href"] != null)
                    {
                        valuationPriceFinalIDRef_ = valuationPriceFinalNode.Attributes["href"].Value;
                    }
                    else
                    {
                        valuationPriceFinal_ = new ReturnLegValuationPrice(valuationPriceFinalNode);
                    }
                }
                else
                {
                    valuationPriceFinal_ = new ReturnLegValuationPrice(valuationPriceFinalNode);
                }
            }
            
        
            XmlNode paymentDatesNode = xmlNode.SelectSingleNode("paymentDates");
            
            if (paymentDatesNode != null)
            {
                if (paymentDatesNode.Attributes["href"] != null || paymentDatesNode.Attributes["id"] != null) 
                {
                    if (paymentDatesNode.Attributes["id"] != null) 
                    {
                        paymentDatesIDRef_ = paymentDatesNode.Attributes["id"].Value;
                        ReturnSwapPaymentDates ob = new ReturnSwapPaymentDates(paymentDatesNode);
                        IDManager.SetID(paymentDatesIDRef_, ob);
                    }
                    else if (paymentDatesNode.Attributes["href"] != null)
                    {
                        paymentDatesIDRef_ = paymentDatesNode.Attributes["href"].Value;
                    }
                    else
                    {
                        paymentDates_ = new ReturnSwapPaymentDates(paymentDatesNode);
                    }
                }
                else
                {
                    paymentDates_ = new ReturnSwapPaymentDates(paymentDatesNode);
                }
            }
            
        
            XmlNode exchangeTradedContractNearestNode = xmlNode.SelectSingleNode("exchangeTradedContractNearest");
            
            if (exchangeTradedContractNearestNode != null)
            {
                if (exchangeTradedContractNearestNode.Attributes["href"] != null || exchangeTradedContractNearestNode.Attributes["id"] != null) 
                {
                    if (exchangeTradedContractNearestNode.Attributes["id"] != null) 
                    {
                        exchangeTradedContractNearestIDRef_ = exchangeTradedContractNearestNode.Attributes["id"].Value;
                        ExchangeTradedContract ob = new ExchangeTradedContract(exchangeTradedContractNearestNode);
                        IDManager.SetID(exchangeTradedContractNearestIDRef_, ob);
                    }
                    else if (exchangeTradedContractNearestNode.Attributes["href"] != null)
                    {
                        exchangeTradedContractNearestIDRef_ = exchangeTradedContractNearestNode.Attributes["href"].Value;
                    }
                    else
                    {
                        exchangeTradedContractNearest_ = new ExchangeTradedContract(exchangeTradedContractNearestNode);
                    }
                }
                else
                {
                    exchangeTradedContractNearest_ = new ExchangeTradedContract(exchangeTradedContractNearestNode);
                }
            }
            
        
        }
        
    
        #region InitialPrice_
        private ReturnLegValuationPrice initialPrice_;
        public ReturnLegValuationPrice InitialPrice_
        {
            get
            {
                if (this.initialPrice_ != null)
                {
                    return this.initialPrice_; 
                }
                else if (this.initialPriceIDRef_ != null)
                {
                    initialPrice_ = IDManager.getID(initialPriceIDRef_) as ReturnLegValuationPrice;
                    return this.initialPrice_; 
                }
                else
                {
                      return this.initialPrice_; 
                }
            }
            set
            {
                if (this.initialPrice_ != value)
                {
                    this.initialPrice_ = value;
                }
            }
        }
        #endregion
        
        public string initialPriceIDRef_ { get; set; }
        #region NotionalReset_
        private XsdTypeBoolean notionalReset_;
        public XsdTypeBoolean NotionalReset_
        {
            get
            {
                if (this.notionalReset_ != null)
                {
                    return this.notionalReset_; 
                }
                else if (this.notionalResetIDRef_ != null)
                {
                    notionalReset_ = IDManager.getID(notionalResetIDRef_) as XsdTypeBoolean;
                    return this.notionalReset_; 
                }
                else
                {
                      return this.notionalReset_; 
                }
            }
            set
            {
                if (this.notionalReset_ != value)
                {
                    this.notionalReset_ = value;
                }
            }
        }
        #endregion
        
        public string notionalResetIDRef_ { get; set; }
        #region ValuationPriceInterim_
        private ReturnLegValuationPrice valuationPriceInterim_;
        public ReturnLegValuationPrice ValuationPriceInterim_
        {
            get
            {
                if (this.valuationPriceInterim_ != null)
                {
                    return this.valuationPriceInterim_; 
                }
                else if (this.valuationPriceInterimIDRef_ != null)
                {
                    valuationPriceInterim_ = IDManager.getID(valuationPriceInterimIDRef_) as ReturnLegValuationPrice;
                    return this.valuationPriceInterim_; 
                }
                else
                {
                      return this.valuationPriceInterim_; 
                }
            }
            set
            {
                if (this.valuationPriceInterim_ != value)
                {
                    this.valuationPriceInterim_ = value;
                }
            }
        }
        #endregion
        
        public string valuationPriceInterimIDRef_ { get; set; }
        #region ValuationPriceFinal_
        private ReturnLegValuationPrice valuationPriceFinal_;
        public ReturnLegValuationPrice ValuationPriceFinal_
        {
            get
            {
                if (this.valuationPriceFinal_ != null)
                {
                    return this.valuationPriceFinal_; 
                }
                else if (this.valuationPriceFinalIDRef_ != null)
                {
                    valuationPriceFinal_ = IDManager.getID(valuationPriceFinalIDRef_) as ReturnLegValuationPrice;
                    return this.valuationPriceFinal_; 
                }
                else
                {
                      return this.valuationPriceFinal_; 
                }
            }
            set
            {
                if (this.valuationPriceFinal_ != value)
                {
                    this.valuationPriceFinal_ = value;
                }
            }
        }
        #endregion
        
        public string valuationPriceFinalIDRef_ { get; set; }
        #region PaymentDates_
        private ReturnSwapPaymentDates paymentDates_;
        public ReturnSwapPaymentDates PaymentDates_
        {
            get
            {
                if (this.paymentDates_ != null)
                {
                    return this.paymentDates_; 
                }
                else if (this.paymentDatesIDRef_ != null)
                {
                    paymentDates_ = IDManager.getID(paymentDatesIDRef_) as ReturnSwapPaymentDates;
                    return this.paymentDates_; 
                }
                else
                {
                      return this.paymentDates_; 
                }
            }
            set
            {
                if (this.paymentDates_ != value)
                {
                    this.paymentDates_ = value;
                }
            }
        }
        #endregion
        
        public string paymentDatesIDRef_ { get; set; }
        #region ExchangeTradedContractNearest_
        private ExchangeTradedContract exchangeTradedContractNearest_;
        public ExchangeTradedContract ExchangeTradedContractNearest_
        {
            get
            {
                if (this.exchangeTradedContractNearest_ != null)
                {
                    return this.exchangeTradedContractNearest_; 
                }
                else if (this.exchangeTradedContractNearestIDRef_ != null)
                {
                    exchangeTradedContractNearest_ = IDManager.getID(exchangeTradedContractNearestIDRef_) as ExchangeTradedContract;
                    return this.exchangeTradedContractNearest_; 
                }
                else
                {
                      return this.exchangeTradedContractNearest_; 
                }
            }
            set
            {
                if (this.exchangeTradedContractNearest_ != value)
                {
                    this.exchangeTradedContractNearest_ = value;
                }
            }
        }
        #endregion
        
        public string exchangeTradedContractNearestIDRef_ { get; set; }
        
    
        
    
    }
    
}

File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-eq-shared-5-4/ReturnLegValuation.cpp
// ReturnLegValuation.cpp 
#include "ReturnLegValuation.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

ReturnLegValuation::ReturnLegValuation(TiXmlNode* xmlNode)
: ISerialized(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //initialPriceNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* initialPriceNode = xmlNode->FirstChildElement("initialPrice");

   if(initialPriceNode){initialPriceIsNull_ = false;}
   else{initialPriceIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- initialPriceNode , address : " << initialPriceNode << std::endl;
   #endif
   if(initialPriceNode)
   {
      if(initialPriceNode->Attribute("href") || initialPriceNode->Attribute("id"))
      {
          if(initialPriceNode->Attribute("id"))
          {
             initialPriceIDRef_ = initialPriceNode->Attribute("id");
             initialPrice_ = boost::shared_ptr<ReturnLegValuationPrice>(new ReturnLegValuationPrice(initialPriceNode));
             initialPrice_->setID(initialPriceIDRef_);
             IDManager::instance().setID(initialPriceIDRef_,initialPrice_);
          }
          else if(initialPriceNode->Attribute("href")) { initialPriceIDRef_ = initialPriceNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { initialPrice_ = boost::shared_ptr<ReturnLegValuationPrice>(new ReturnLegValuationPrice(initialPriceNode));}
   }

   //notionalResetNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* notionalResetNode = xmlNode->FirstChildElement("notionalReset");

   if(notionalResetNode){notionalResetIsNull_ = false;}
   else{notionalResetIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- notionalResetNode , address : " << notionalResetNode << std::endl;
   #endif
   if(notionalResetNode)
   {
      if(notionalResetNode->Attribute("href") || notionalResetNode->Attribute("id"))
      {
          if(notionalResetNode->Attribute("id"))
          {
             notionalResetIDRef_ = notionalResetNode->Attribute("id");
             notionalReset_ = boost::shared_ptr<XsdTypeBoolean>(new XsdTypeBoolean(notionalResetNode));
             notionalReset_->setID(notionalResetIDRef_);
             IDManager::instance().setID(notionalResetIDRef_,notionalReset_);
          }
          else if(notionalResetNode->Attribute("href")) { notionalResetIDRef_ = notionalResetNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { notionalReset_ = boost::shared_ptr<XsdTypeBoolean>(new XsdTypeBoolean(notionalResetNode));}
   }

   //valuationPriceInterimNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* valuationPriceInterimNode = xmlNode->FirstChildElement("valuationPriceInterim");

   if(valuationPriceInterimNode){valuationPriceInterimIsNull_ = false;}
   else{valuationPriceInterimIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- valuationPriceInterimNode , address : " << valuationPriceInterimNode << std::endl;
   #endif
   if(valuationPriceInterimNode)
   {
      if(valuationPriceInterimNode->Attribute("href") || valuationPriceInterimNode->Attribute("id"))
      {
          if(valuationPriceInterimNode->Attribute("id"))
          {
             valuationPriceInterimIDRef_ = valuationPriceInterimNode->Attribute("id");
             valuationPriceInterim_ = boost::shared_ptr<ReturnLegValuationPrice>(new ReturnLegValuationPrice(valuationPriceInterimNode));
             valuationPriceInterim_->setID(valuationPriceInterimIDRef_);
             IDManager::instance().setID(valuationPriceInterimIDRef_,valuationPriceInterim_);
          }
          else if(valuationPriceInterimNode->Attribute("href")) { valuationPriceInterimIDRef_ = valuationPriceInterimNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { valuationPriceInterim_ = boost::shared_ptr<ReturnLegValuationPrice>(new ReturnLegValuationPrice(valuationPriceInterimNode));}
   }

   //valuationPriceFinalNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* valuationPriceFinalNode = xmlNode->FirstChildElement("valuationPriceFinal");

   if(valuationPriceFinalNode){valuationPriceFinalIsNull_ = false;}
   else{valuationPriceFinalIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- valuationPriceFinalNode , address : " << valuationPriceFinalNode << std::endl;
   #endif
   if(valuationPriceFinalNode)
   {
      if(valuationPriceFinalNode->Attribute("href") || valuationPriceFinalNode->Attribute("id"))
      {
          if(valuationPriceFinalNode->Attribute("id"))
          {
             valuationPriceFinalIDRef_ = valuationPriceFinalNode->Attribute("id");
             valuationPriceFinal_ = boost::shared_ptr<ReturnLegValuationPrice>(new ReturnLegValuationPrice(valuationPriceFinalNode));
             valuationPriceFinal_->setID(valuationPriceFinalIDRef_);
             IDManager::instance().setID(valuationPriceFinalIDRef_,valuationPriceFinal_);
          }
          else if(valuationPriceFinalNode->Attribute("href")) { valuationPriceFinalIDRef_ = valuationPriceFinalNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { valuationPriceFinal_ = boost::shared_ptr<ReturnLegValuationPrice>(new ReturnLegValuationPrice(valuationPriceFinalNode));}
   }

   //paymentDatesNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* paymentDatesNode = xmlNode->FirstChildElement("paymentDates");

   if(paymentDatesNode){paymentDatesIsNull_ = false;}
   else{paymentDatesIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- paymentDatesNode , address : " << paymentDatesNode << std::endl;
   #endif
   if(paymentDatesNode)
   {
      if(paymentDatesNode->Attribute("href") || paymentDatesNode->Attribute("id"))
      {
          if(paymentDatesNode->Attribute("id"))
          {
             paymentDatesIDRef_ = paymentDatesNode->Attribute("id");
             paymentDates_ = boost::shared_ptr<ReturnSwapPaymentDates>(new ReturnSwapPaymentDates(paymentDatesNode));
             paymentDates_->setID(paymentDatesIDRef_);
             IDManager::instance().setID(paymentDatesIDRef_,paymentDates_);
          }
          else if(paymentDatesNode->Attribute("href")) { paymentDatesIDRef_ = paymentDatesNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { paymentDates_ = boost::shared_ptr<ReturnSwapPaymentDates>(new ReturnSwapPaymentDates(paymentDatesNode));}
   }

   //exchangeTradedContractNearestNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* exchangeTradedContractNearestNode = xmlNode->FirstChildElement("exchangeTradedContractNearest");

   if(exchangeTradedContractNearestNode){exchangeTradedContractNearestIsNull_ = false;}
   else{exchangeTradedContractNearestIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- exchangeTradedContractNearestNode , address : " << exchangeTradedContractNearestNode << std::endl;
   #endif
   if(exchangeTradedContractNearestNode)
   {
      if(exchangeTradedContractNearestNode->Attribute("href") || exchangeTradedContractNearestNode->Attribute("id"))
      {
          if(exchangeTradedContractNearestNode->Attribute("id"))
          {
             exchangeTradedContractNearestIDRef_ = exchangeTradedContractNearestNode->Attribute("id");
             exchangeTradedContractNearest_ = boost::shared_ptr<ExchangeTradedContract>(new ExchangeTradedContract(exchangeTradedContractNearestNode));
             exchangeTradedContractNearest_->setID(exchangeTradedContractNearestIDRef_);
             IDManager::instance().setID(exchangeTradedContractNearestIDRef_,exchangeTradedContractNearest_);
          }
          else if(exchangeTradedContractNearestNode->Attribute("href")) { exchangeTradedContractNearestIDRef_ = exchangeTradedContractNearestNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { exchangeTradedContractNearest_ = boost::shared_ptr<ExchangeTradedContract>(new ExchangeTradedContract(exchangeTradedContractNearestNode));}
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<ReturnLegValuationPrice> ReturnLegValuation::getInitialPrice()
{
   if(!this->initialPriceIsNull_){
        if(initialPriceIDRef_ != ""){
             return boost::shared_static_cast<ReturnLegValuationPrice>(IDManager::instance().getID(initialPriceIDRef_));
        }else{
             return this->initialPrice_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<ReturnLegValuationPrice>();
   }
}
boost::shared_ptr<XsdTypeBoolean> ReturnLegValuation::getNotionalReset()
{
   if(!this->notionalResetIsNull_){
        if(notionalResetIDRef_ != ""){
             return boost::shared_static_cast<XsdTypeBoolean>(IDManager::instance().getID(notionalResetIDRef_));
        }else{
             return this->notionalReset_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<XsdTypeBoolean>();
   }
}
boost::shared_ptr<ReturnLegValuationPrice> ReturnLegValuation::getValuationPriceInterim()
{
   if(!this->valuationPriceInterimIsNull_){
        if(valuationPriceInterimIDRef_ != ""){
             return boost::shared_static_cast<ReturnLegValuationPrice>(IDManager::instance().getID(valuationPriceInterimIDRef_));
        }else{
             return this->valuationPriceInterim_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<ReturnLegValuationPrice>();
   }
}
boost::shared_ptr<ReturnLegValuationPrice> ReturnLegValuation::getValuationPriceFinal()
{
   if(!this->valuationPriceFinalIsNull_){
        if(valuationPriceFinalIDRef_ != ""){
             return boost::shared_static_cast<ReturnLegValuationPrice>(IDManager::instance().getID(valuationPriceFinalIDRef_));
        }else{
             return this->valuationPriceFinal_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<ReturnLegValuationPrice>();
   }
}
boost::shared_ptr<ReturnSwapPaymentDates> ReturnLegValuation::getPaymentDates()
{
   if(!this->paymentDatesIsNull_){
        if(paymentDatesIDRef_ != ""){
             return boost::shared_static_cast<ReturnSwapPaymentDates>(IDManager::instance().getID(paymentDatesIDRef_));
        }else{
             return this->paymentDates_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<ReturnSwapPaymentDates>();
   }
}
boost::shared_ptr<ExchangeTradedContract> ReturnLegValuation::getExchangeTradedContractNearest()
{
   if(!this->exchangeTradedContractNearestIsNull_){
        if(exchangeTradedContractNearestIDRef_ != ""){
             return boost::shared_static_cast<ExchangeTradedContract>(IDManager::instance().getID(exchangeTradedContractNearestIDRef_));
        }else{
             return this->exchangeTradedContractNearest_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<ExchangeTradedContract>();
   }
}
}

